Github/Confusing Things in Git Terminology.md

```diff
@@ -0,0 +1,64 @@
+## Confusing Terms in the Git Terminology
+
+Git and GitHub are at the core of open source in today's scenario. However, there are a lot of confusing terms that often seem similar yet can have conflicting meanings and uses. Let's eradicate these confusing terms by decoding their underlying meanings and sense.
+
+- ### Origin and Upstream
+
+From Git Documentation:
+> When a repo is cloned, it has a default remote called origin that points to your fork on GitHub, not the original repo it was forked from.
+
+>To keep track of the original repo, you need to add another remote named upstream
+
+This confusion arises when you are working on the clone of a forked repo. **Upstream** is what you would use to **fetch from the original repo** in order to keep your local copy in sync with the project you wanted to contributed to. Meanwhile, **Origin** is what you would use to **pull and push code to your own fork** and thereafter you can contribute back to the upstream repo by making a pull request.
+
+![Upstream and origin](https://i2.wp.com/i.stack.imgur.com/L5wRU.png)
+source: [https://stackoverflow.com](https://stackoverflow.com)
+
+**Remember**: Remotes are simply an alias that store the URL of repositories.
+
+- ### Fetch and Pull
+
+From Git Documentation:
+> git fetch can fetch from either a single named repository or URL, or from several repositories at once.
+
+>git pull incorporates changes from a remote repository into the current branch.
+
+**git fetch** is the command that tells your local git to retrieve the latest meta-data info from the original yet doesn’t do any file transferring. It’s more like just checking to see if there are any changes available. **git pull** on the other hand does that AND copies those changes from the remote repository.
+
+![fetch vs pull](https://i.stack.imgur.com/nWYnQ.png)
+source: [https://stackoverflow.com](https://stackoverflow.com)
+
+**Remember**:  `git pull` is shorthand for `git fetch` followed by `git merge FETCH_HEAD`
+
+Putting it rather simply, **git fetch** can be used to know the changes done in the remote repo/branch since your last pull. This is useful to allow for checking before doing a **git pull**, which could change files in your current branch and working copy and potentially lose your changes, etc.
```

Comment:
I use git fetch to get names of branches on origin so that I can checkout to those branches locally. What are other benefits of `git fetch`? giving more concrete examples would be great.